lib/registry.dart

```diff
@@ -260,17 +260,17 @@ class Registry<T> {
       // Response is even-length list of (id, element) pairs.
       var cache = _cache;
       var count = response.length ~/ 2;
-      var result = List<T>.filled(count, null);
+      var result = List<T?>.filled(count, null);
       for (var i = 0; i < count; i++) {
         var id = response[i * 2] as int;
-        var element = response[i * 2 + 1] as T;
-        element = cache.register(id, element);
+        T? element = response[i * 2 + 1] as T;
```

Comment:
Using `T? element = ... as T;` *looks* wrong. 
I can see its just to enable reusing the variable below, but that suggests that the sharing is no longer meaningful when the types have diverged. I'd rather do:

```dart
var element = response[...] as T;
var registeredElement = cache.register(id, element);
result[i] = registeredElement;
```

Reply:
Sorry but this cannot be inferred.

Reply:
Maybe replace the entire list generation with the following, and change the `(List response)` above to `(List<dynamic> response)`:
```dart
var result = List<T>.generate(
    count, (i) => cache.register(response[i * 2], response[i * 2 + 1]) as T,
    growable: false);
```

Reply:
And again, the `response[i * 2]` part is still unable to be inferred.
![image](https://user-images.githubusercontent.com/15884415/120284689-17f77e00-c2ef-11eb-84e9-efd418ec89a2.png)

Reply:
Unless we force cast the `response[i * 2]` to int, WDYT?

Reply:
That's why I wanted to type `response` as `List<dynamic>`, to get the implicit downcast.
The alternative is to add the `as int` and `as Object` to the arguments. Explicit is probably better, even though the code is generally dynamically typed, so `dynamic` would fit in.

Reply:
> That's why I wanted to type `response` as `List<dynamic>`

![image](https://user-images.githubusercontent.com/15884415/120289062-7888ba00-c2f3-11eb-9786-07472a51663b.png)

So I guess we can only implement explicit declarations now.